IOPool/Output/src/RootOutputFile.cc

```diff
@@ -815,13 +815,21 @@ namespace edm {
                                         SelectedProducts const& branches,
                                         std::string const& processName) const {
     if (tree && tree->GetNbranches() != 0) {
+      auto const& aliasForBranches = om_->aliasForBranches();
       for (auto const& selection : branches) {
         ProductDescription const& pd = *selection.first;
         if (pd.branchType() == InProcess && processName != pd.processName()) {
           continue;
         }
         std::string const& full = pd.branchName() + "obj";
-        if (pd.branchAliases().empty()) {
+        bool matched = false;
+        for (auto const& matcher : aliasForBranches) {
+          if (matcher.match(pd.branchName())) {
+            tree->SetAlias(matcher.alias_.c_str(), full.c_str());
+            matched = true;
+          }
+        }
+        if (not matched and pd.branchAliases().empty()) {
           std::string const& alias = (pd.productInstanceName().empty() ? pd.moduleLabel() : pd.productInstanceName());
           tree->SetAlias(alias.c_str(), full.c_str());
         } else {
```

Comment:
If there is a match in `aliasForBranches`, should the `else` branch be skipped as well?

Reply:
If `pd.branchAliases().empty()` is false it means there were hard coded `addAlias` calls in the EDProducer. I'd rather not cause those to fail yet.

Reply:
Ok